OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -106,26 +106,40 @@ void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barre
 
 		void INotifyAttack.PreparingAttack(Actor self, Target target, Armament a, Barrel barrel) { }
 
-		void ITick.Tick(Actor self)
+		public virtual void Reload(Actor self, int reloadDelay, int reloadCount, string sound, bool needsToBeDocked, bool resetOnFire = false)
 		{
-			if (!Info.SelfReloads)
-				return;
-
 			// Resets the tick counter if ammo was fired.
-			if (Info.ResetOnFire && GetAmmoCount() < PreviousAmmo)
+			if (resetOnFire && GetAmmoCount() < PreviousAmmo)
 			{
-				RemainingTicks = Info.SelfReloadDelay;
+				RemainingTicks = reloadDelay;
 				PreviousAmmo = GetAmmoCount();
 			}
 
 			if (!FullAmmo() && --RemainingTicks == 0)
 			{
-				RemainingTicks = Info.SelfReloadDelay;
+				// TODO: Make this configurable or replace it with something better entirely
```

Comment:
With my request above this would stay in `Rearm` for now, and could be dropped completely in favor of a pause/unpause condition on `ReloadAmmoPool` that is activated by the docking trait.